services/github/github-package-registry-downloads.service.js

```diff
@@ -0,0 +1,295 @@
+    if (version) {
+      if (version === 'latest') {
+        return await this._requestGraphql({
+          query: gql`
+            query($user: String!, $repo: String!, $packageName: String!) {
+              repository(owner: $user, name: $repo) {
+                registryPackages(name: $packageName, first: 1) {
+                  nodes {
+                    latestVersion {
+                      statistics {
+                        downloadsThisMonth
+                        downloadsThisWeek
+                        downloadsThisYear
+                        downloadsToday
+                        downloadsTotalCount
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          `,
+          variables: { user, repo, packageName },
+          schema: latestVersionDownloadsSchema,
+          transformErrors,
+        })
+      } else {
+        return await this._requestGraphql({
+          query: gql`
+            query(
+              $user: String!
+              $repo: String!
+              $packageName: String!
+              $version: String!
+            ) {
+              repository(owner: $user, name: $repo) {
+                registryPackages(name: $packageName, first: 1) {
+                  nodes {
+                    version(version: $version) {
+                      statistics {
+                        downloadsThisMonth
+                        downloadsThisWeek
+                        downloadsThisYear
+                        downloadsToday
+                        downloadsTotalCount
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          `,
+          variables: { user, repo, packageName, version },
+          schema: versionDownloadsSchema,
+          transformErrors,
+        })
+      }
+    } else {
+      return await this._requestGraphql({
+        query: gql`
+          query($user: String!, $repo: String!, $packageName: String!) {
+            repository(owner: $user, name: $repo) {
+              registryPackages(name: $packageName, first: 1) {
+                nodes {
+                  statistics {
+                    downloadsThisMonth
+                    downloadsThisWeek
+                    downloadsThisYear
+                    downloadsToday
+                    downloadsTotalCount
+                  }
+                }
+              }
+            }
+          }
+        `,
+        variables: { user, repo, packageName },
+        schema: totalDownloadsSchema,
+        transformErrors,
+      })
+    }
+  }
+
+  transform({ json, interval, version }) {
+    const ghPackage = json.data.repository.registryPackages.nodes[0]
+    if (!ghPackage) {
+      throw new NotFound({ prettyMessage: 'package not found' })
+    }
+
+    let statistics
+    if (version === 'latest') {
+      statistics = ghPackage.latestVersion.statistics
```

Comment:
AFAICT, if a package exists then it by definition must have a `latestVersion`, so I'm not worried about any `falsy` values for `latestVersion` here